test/test_datasets_download.py

```diff
@@ -149,10 +154,12 @@ def retry(fn, times=1, wait=5.0):
 def assert_server_response_ok():
     try:
         yield
-    except URLError as error:
-        raise AssertionError("The request timed out.") from error
     except HTTPError as error:
         raise AssertionError(f"The server returned {error.code}: {error.reason}.") from error
+    except URLError as error:
```

Comment:
1. `assert issubclass(HTTPError, URLError)` and thus we need to catch it first or otherwise the branch is unreachable.
2. SSL issues also raise an `URLError` which is not a timing out request.